### PR TITLE
Merge master to stable

### PR DESCRIPTION
**Pull request by geoffnichols**
This PR merges master to stable to include changes needed for EL and SLES s390x platforms.

### CODE REVIEW

**Comment by geoffnichols** [comment]
Doing a test build with this now...

**Comment by geoffnichols** [comment]
Test build with this branch is currently failing for sles-11-s390.

File: configs/components/augeas.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 component 'augeas' do |pkg, settings, platform|
   pkg.version '1.4.0'
   pkg.md5sum 'a2536a9c3d744dc09d234228fe4b0c93'
-  pkg.url "http://buildsources.delivery.puppetlabs.net/#{pkg.get_name}-#{pkg.get_version}.tar.gz"
+  pkg.url "http://download.augeas.net/augeas-#{pkg.get_version}.tar.gz"
```
Comment (McdonaldSeanp) [inline configs/components/augeas.rb]: @stahnma I'm just triple checking here, we absolutely want the upstream stuff with the rewrites for the next release?

**Comment by McdonaldSeanp** [comment]
This all looks 👍 from my standpoint, none of these windows changes will effect the current pftw work, and if we want to ship puppet-agent with vanagon on windows for the next release we will need those changes.

File: configs/components/augeas.rb (hunk 1 above)
Comment (stahnma) [inline configs/components/augeas.rb]: yes

**Comment by geoffnichols** [comment]
Test build (two links because needed to matrix reload) was successful:

http://jenkins-release.delivery.puppetlabs.net/job/vanagon_generic_job/482/
http://jenkins-release.delivery.puppetlabs.net/job/vanagon_generic_job/483/

**Pull request merged**